### ISSUE TITLE
Optimize bundle size

### ISSUE THREAD

**Issue by atomfrede** [jhipster/jhipster-vuejs#129]
As already mentioned in #94 moment is a large monster. I have run the bundle analyzer and found a bit of potential optimizations.

* First the tree shaking of fontawesome seems not to work as expected (again, we had the same issues with react/angular)
* There are currently all moment locales which are in that concrete case are not required (just selected `en` no other language)

* What seems to work much better is splitting of our main application into separate modules (see the blue parts on the right)
* again good to see how small vue is compared to angularX

![screenshot_2018-12-17 webpack bundle analyzer](https://user-images.githubusercontent.com/203401/50113606-bf2ceb80-0242-11e9-98fd-f99c5bce1854.png)

**Comment by atomfrede** [comment]
Just to track the progress here, after using a custom icon lib the result looks like this:

![50114693-74f93980-0245-11e9-95b8-9ac93d80068c](https://user-images.githubusercontent.com/203401/50162545-f0f39000-02de-11e9-8e4e-4910ac94a469.png)

**Comment by atomfrede** [comment]
Here is the result using date-fns with a couple of languages (~1.9mb). I wonder if we could further optimize it by e.g. changing the way we handle i18n changes. In case we could accept a hard page reload when changing the language we might just load one particular locale instead of all supported locales.

![screenshot_2018-12-18 webpack bundle analyzer](https://user-images.githubusercontent.com/203401/50181945-33809100-030e-11e9-9f8c-762431b7912e.png)

**Comment by sahbi-ktifa** [comment]
I've started to change bootstrap-vue components loading as we currently import everything. I'll PR that soon.

**Comment by atomfrede** [comment]
@sahbi-ktifa Great, looking forward to it!

**Comment by sahbi-ktifa** [comment]
I'm a bit disappointed cause if I only use separated bootstrap vue components, my bundle is bigger... I'm still confused about this and I must be doing something wrong here. Any idea @atomfrede ?

**Comment by atomfrede** [comment]
Not sure. What did you try? Do we need to configure something in the webpack config?

**Comment by sahbi-ktifa** [comment]
I only did what the doc said, instead of loading all bootstrap-vue, I used only required components. My bundle was bigger afterwards... I surely missed something, I'll take another shot soon

**Comment by atomfrede** [comment]
Indeed I am confused when looking at their docs what to do exactly

**Comment by wmarques** [comment]
@atomfrede I wonder if we can use dynamic imports to dynamically load the language file, this can reduce bundle size.

**Comment by atomfrede** [comment]
Yes that would be something I would like to try after we have date-fns integrated. Hope to continue later today.

**Comment by sahbi-ktifa** [comment]
I successfully loaded required only components from bootstrap, check #154

**Comment by deepu105** [comment]
You guys are doing awesome :)

Thanks & Regards,
Deepu

**Comment by atomfrede** [comment]
I am nearly done with date-fns integration and have for a sample app without any entities we have gzipped size for all js files of `173.32 KB`. Removing the additional ~30kb from the bootstrap parts results in ~`143KB` which seems quite nice.

**Comment by atomfrede** [comment]
@mraible This is interesting for you. Simple app with defaults and one entity. It feels just blazing fast 

https://datefns.herokuapp.com/#/

![screenshot-20181221222933-993x916](https://user-images.githubusercontent.com/203401/50364356-fddae280-056f-11e9-92a8-cf9476b276c8.png)

**Comment by atomfrede** [comment]
After integration the bootstrap branch we have a final bundle size of `154.94 KB` gzipped. 
![screenshot_2018-12-22 webpack bundle analyzer](https://user-images.githubusercontent.com/203401/50377249-1f90a400-061a-11e9-9b03-35999b4989bc.png)

**Comment by hdurix** [comment]
Awesome @atomfrede!

What was the initial size before your optimization work? And do you know the bundle size for react & angular?

**Comment by jdubois** [comment]
Awesome work and well-earned bug bounty! @atomfrede don't forget we are in the bug bounty booster program, so that's $200. Merry Christmas 😀

**Comment by atomfrede** [comment]
Sadly I didn't measure it before, but I think it was ~1.x mb (gzipped, mostly due to all font awesome icons). It can check how large our react and angular parts are ( I did that a few month ago already, maybe I find the screenshot somewhere).

@sahbi-ktifa How to we handle the bounty btw?

**Comment by atomfrede** [comment]
@jdubois So I just file 200 instead of 100?

**Comment by jdubois** [comment]
@atomfrede yes it's the usual process but you invoice $200 instead of $100 :-)
(Oh don't worry for our finances, I've heard we'll soon have a new silver sponsor 😀)

**Comment by atomfrede** [comment]
@hdurix Just generated a sample angular app. The gzipped bundle size is 290kb.
![screenshot_2018-12-22 webpack bundle analyzer](https://user-images.githubusercontent.com/203401/50377764-88c8e500-0623-11e9-807d-3282a3936d1d.png)

**Comment by hdurix** [comment]
Very interesting!

**Comment by sahbi-ktifa** [comment]
@atomfrede as bounty is now 200$, what do you think of 150/50 as my part was lighter than yours?

**Comment by atomfrede** [comment]
Damn, just submitted the expense. @jdubois Can we do it like @sahbi-ktifa suggested? Need to cancel my expense

**Comment by sahbi-ktifa** [comment]
No no don't. Don't worry, I'll get another bounty 😉 @jdubois you can approve @atomfrede expense !
